src/components/NewArticle/NewArticle.tsx

```diff
@@ -0,0 +1,259 @@
+} from 'fp-ts/Option'
+import {
+  type AlertProps,
+  type TextProps,
+  type InputProps,
+  Button,
+  TextButton,
+  TextEditor,
+} from '@/components'
+import { useFormik } from 'formik'
+import { object, string } from 'yup'
+import { useNewArticle } from './useNewArticle'
+import {
+  type AuthorizationError,
+  type UnknownError,
+  type ValidationError,
+  f,
+} from '@/libs'
+import { useEffect, useState } from 'react'
+import styled from 'styled-components'
+import dynamic from 'next/dynamic'
+import {
+  parse as superJsonParse,
+  stringify as superJsonStringify,
+} from 'superjson'
+
+const Alert = {
+  Root: dynamic<AlertProps>(
+    () => import('@/components/Alert/Alert').then((module) => module.Alert),
+    {
+      ssr: false,
+    }
+  ),
+  Text: dynamic<TextProps>(
+    () =>
+      import('@/components/Alert/Alert').then((module) => module.Alert.Text),
+    {
+      ssr: false,
+    }
+  ),
+  Icon: dynamic<unknown>(
+    () =>
+      import('@/components/Alert/Alert').then((module) => module.Alert.Icon),
+    {
+      ssr: false,
+    }
+  ),
+  Close: dynamic<unknown>(
+    () =>
+      import('@/components/Alert/Alert').then((module) => module.Alert.Close),
+    {
+      ssr: false,
+    }
+  ),
+}
+
+const Input = dynamic<InputProps>(
+  () => import('@/components/Input/Input').then((module) => module.Input),
+  { ssr: false }
+)
+
+const Wrapper = styled.form`
+  display: flex;
+  flex-direction: column;
+  row-gap: ${({ theme }) => theme.spacings.xmedium};
+  width: 100%;
+  max-width: 700px;
+`
+
+const Title = styled.h1`
+  font-size: ${({ theme }) => theme.fonts.sizes.huge};
+  color: ${({ theme }) => theme.colors.black[400]};
+`
+
+const FieldWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  row-gap: ${({ theme }) => theme.spacings.xxsmall};
+`
+
+const PublishBtn = styled(Button)`
+  align-self: flex-end;
+`
+
+const AlertContentWrapper = styled.div`
+  width: 100%;
+  display: flex;
+  gap: ${({ theme }) => theme.spacings.xsmall};
+`
+
+const newArticleSchema = object({
+  title: string()
+    .required('Title is a required field and cannot be empty')
+    .max(32, 'Maximum 32 characters required!'),
+  description: string()
+    .required('Description is a required field and cannot be empty')
+    .max(64, 'Maximum 64 characters required!'),
+  body: string().required('Body is a required field and cannot be empty!'),
+})
+
+type NewArticleFieldValues = {
+  title: string
+  description: string
+  body: string
+}
+
+export type NewArticleRequest = {
+  article: NewArticleFieldValues
+}
+
+const initialFieldValues: NewArticleFieldValues = {
+  title: '',
+  description: '',
+  body: '',
+}
+
+const storageKey = 'new-article-editor'
+
+const getStorage = () => {
+  if (typeof window === 'undefined') return initialFieldValues
+
+  const cachedStorage = fromNullable(localStorage.getItem(storageKey))
+
+  return isSome(cachedStorage)
+    ? superJsonParse<NewArticleFieldValues>(cachedStorage.value)
```

Comment:
Validar casos de erro e validar o retorno com io-ts.

Reply:
What do you think about this [way](https://github.com/joaooliveiradev/conduit-frontend/pull/43/commits/8a9c2ead395250545dce964d12e8be95fe3d79a5)